guides/keras_cv/classification_with_keras_cv.py

```diff
@@ -0,0 +1,701 @@
+
+
+def unpackage_dict(inputs):
+    return inputs["images"], inputs["labels"]
+
+
+train_ds = train_ds.map(unpackage_dict, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.map(unpackage_dict, num_parallel_calls=tf.data.AUTOTUNE)
+
+"""
+Data augmentation is by far the hardest piece of training a modern
+classifier.
+Congratulations on making it this far!
+
+## Optimizer Tuning
+
+To achieve optimal performance,we need to use a learning rate schedule instead
+of a single learning rate. While we won't go into detail on the Cosine decay
+with warmup schedule used here, [you can read more about it
+here](https://scorrea92.medium.com/cosine-learning-rate-decay-e8b50aa455b).
+"""
+
+
+def lr_warmup_cosine_decay(
+    global_step,
+    warmup_steps,
+    hold=0,
+    total_steps=0,
+    start_lr=0.0,
+    target_lr=1e-2,
+):
+    # Cosine decay
+    learning_rate = (
+        0.5
+        * target_lr
+        * (
+            1
+            + tf.cos(
+                tf.constant(math.pi)
+                * tf.cast(global_step - warmup_steps - hold, tf.float32)
+                / float(total_steps - warmup_steps - hold)
+            )
+        )
+    )
+
+    warmup_lr = tf.cast(target_lr * (global_step / warmup_steps), tf.float32)
+    target_lr = tf.cast(target_lr, tf.float32)
+
+    if hold > 0:
+        learning_rate = tf.where(
+            global_step > warmup_steps + hold, learning_rate, target_lr
+        )
+
+    learning_rate = tf.where(global_step < warmup_steps, warmup_lr, learning_rate)
+    return learning_rate
+
+
+class WarmUpCosineDecay(schedules.LearningRateSchedule):
+    def __init__(self, warmup_steps, total_steps, hold, start_lr=0.0, target_lr=1e-2):
+        super().__init__()
+        self.start_lr = start_lr
+        self.target_lr = target_lr
+        self.warmup_steps = warmup_steps
+        self.total_steps = total_steps
+        self.hold = hold
+
+    def __call__(self, step):
+        lr = lr_warmup_cosine_decay(
+            global_step=step,
+            total_steps=self.total_steps,
+            warmup_steps=self.warmup_steps,
+            start_lr=self.start_lr,
+            target_lr=self.target_lr,
+            hold=self.hold,
+        )
+
+        return tf.where(step > self.total_steps, 0.0, lr, name="learning_rate")
+
+
+"""
+![WarmUpCosineDecay schedule](https://i.imgur.com/YCr5pII.png)
+
+The schedule looks a as we expect.
+
+Next let's construct this optimizer:
+"""
+
+total_images = 9000
+total_steps = (total_images // BATCH_SIZE) * EPOCHS
+warmup_steps = int(0.1 * total_steps)
+hold_steps = int(0.45 * total_steps)
+schedule = WarmUpCosineDecay(
+    start_lr=0.05,
+    target_lr=1e-2,
+    warmup_steps=warmup_steps,
+    total_steps=total_steps,
+    hold=hold_steps,
+)
+optimizer = optimizers.SGD(
+    decay=5e-4,
+    learning_rate=schedule,
+    momentum=0.9,
+)
+"""
+At long last, we can now build our model and call `fit()`!
+"""
+
+backbone = keras_cv.models.EfficientNetV2Backbone.from_preset(
```

Comment:
We probably should have some comment about this preset not coming with pretrained weights. Alternatively, this would be a good opportunity to introduce our alias models (i.e., `keras_cv.models.EfficientNetV2B0Backbone()`)